bahmni_sale/models/sale_order_line.py

```diff
@@ -54,8 +69,13 @@ def get_available_batch_details(self, product_id, sale_order):
                  ('id', 'not in', already_used_batch_ids if already_used_batch_ids else False)]\
                  if len(already_used_batch_ids) > 0 else [('product_id','=', product_id.id if type(product_id) != list else product_id[0])]
                  
-        for prodlot in stock_prod_lot.search(query, order='expiration_date asc'):
-            if(prodlot.expiration_date and datetime.strptime(str(prodlot.expiration_date), DTF) > datetime.today()):
+        for prodlot in stock_prod_lot.search(query, order='expiration_date asc'):            
+            date_lenth = len(str(prodlot.expiration_date))            
+            if len(str(prodlot.expiration_date)) > 20:                
```

Comment:
why is this length check needed for a date field ?

Reply:
@mohan-13 , Its date format MS will added. So, Auto lot selection date convert not working. Will remove and convert the date logic.